sormas-backend/src/main/java/de/symeda/sormas/backend/action/ActionService.java

```diff
@@ -54,18 +57,68 @@ public class ActionService extends AdoServiceWithUserFilter<Action> {
 	@EJB
 	private EventService eventService;
 
+	@EJB
+	private DocumentService documentService;
+
 	public ActionService() {
 		super(Action.class);
 	}
 
-	public List<Action> getAllActionsAfter(Date date, User user) {
+	public List<Action> getAllAfter(Date since, Integer batchSize, String lastSynchronizedUuid) {
 
 		CriteriaBuilder cb = em.getCriteriaBuilder();
 		CriteriaQuery<Action> cq = cb.createQuery(getElementClass());
+		Root<Action> root = cq.from(getElementClass());
+
+		ActionQueryContext queryContext = new ActionQueryContext(cb, cq, root);
+
+		Predicate filter = createActiveFilter(cb, queryContext.getJoins().getEvent());
+		filter = CriteriaBuilderHelper.and(cb, filter, createUserFilter(queryContext));
+		if (since != null) {
+			Predicate dateFilter = createChangeDateFilter(cb, root, since, lastSynchronizedUuid);
+			if (filter != null) {
```

Comment:
Use CriteriaBuilderHelper.and here as well instead of the if-else condition; also allows you to get rid of the variable definition for dateFilter